CI broken on s390x - toolchain issue

As of Nov 1st, CI runs on s390x fail with:
```

2022-11-02T07:14:01.8673501Z   -- Check for working C compiler: /usr/bin/clang-9
2022-11-02T07:14:02.0540808Z   -- Check for working C compiler: /usr/bin/clang-9 - broken
2022-11-02T07:14:02.0560977Z   CMake Error at /usr/share/cmake-3.24/Modules/CMakeTestCCompiler.cmake:69 (message):
2022-11-02T07:14:02.0562007Z     The C compiler
2022-11-02T07:14:02.0562604Z   
2022-11-02T07:14:02.0563632Z       "/usr/bin/clang-9"
2022-11-02T07:14:02.0564380Z   
2022-11-02T07:14:02.0599287Z     is not able to compile a simple test program.
2022-11-02T07:14:02.0600010Z   
2022-11-02T07:14:02.0600923Z     It fails with the following output:
2022-11-02T07:14:02.0601561Z   
2022-11-02T07:14:02.0602268Z       Change Dir: /__w/1/s/artifacts/obj/mono/Linux.s390x.Release/CMakeFiles/CMakeTmp
2022-11-02T07:14:02.0603631Z       
2022-11-02T07:14:02.0605647Z       Run Build Command(s):/usr/bin/make -f Makefile cmTC_f8f53/fast && /usr/bin/make  -f CMakeFiles/cmTC_f8f53.dir/build.make CMakeFiles/cmTC_f8f53.dir/build
2022-11-02T07:14:02.0607679Z       make[1]: Entering directory '/__w/1/s/artifacts/obj/mono/Linux.s390x.Release/CMakeFiles/CMakeTmp'
2022-11-02T07:14:02.0608717Z       Building C object CMakeFiles/cmTC_f8f53.dir/testCCompiler.c.o
2022-11-02T07:14:02.0611975Z       /usr/bin/clang-9 --target=s390x-linux-gnu --gcc-toolchain=/crossrootfs/s390x/usr --sysroot=/crossrootfs/s390x   -Wl,--build-id=sha1  -MD -MT CMakeFiles/cmTC_f8f53.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_f8f53.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_f8f53.dir/testCCompiler.c.o -c /__w/1/s/artifacts/obj/mono/Linux.s390x.Release/CMakeFiles/CMakeTmp/testCCompiler.c
2022-11-02T07:14:02.0615961Z       clang: warning: -Wl,--build-id=sha1: 'linker' input unused [-Wunused-command-line-argument]
2022-11-02T07:14:02.0616914Z       Linking C executable cmTC_f8f53
2022-11-02T07:14:02.0618932Z       /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_f8f53.dir/link.txt --verbose=1
2022-11-02T07:14:02.0622391Z       /usr/bin/clang-9 --target=s390x-linux-gnu --gcc-toolchain=/crossrootfs/s390x/usr --sysroot=/crossrootfs/s390x  -Wl,--build-id=sha1  -Wl,--rpath-link=/crossrootfs/s390x/lib/s390x-linux-gnu -Wl,--rpath-link=/crossrootfs/s390x/usr/lib/s390x-linux-gnu -Wl,--rpath-link=/crossrootfs/s390x/lib/s390x-linux-gnu -Wl,--rpath-link=/crossrootfs/s390x/usr/lib/s390x-linux-gnu -fuse-ld=lld  CMakeFiles/cmTC_f8f53.dir/testCCompiler.c.o -o cmTC_f8f53 
2022-11-02T07:14:02.0624645Z       ld.lld: error: unknown emulation: elf64_s390
```

On s390x, lld is not supported, so it should not be used.  I'm not sure why it is now trying to do that.

@directhex @steveisok @akoeplinger did anything change on the build servers recently?

--- a/eng/common/native/init-compiler.sh
+++ b/eng/common/native/init-compiler.sh
@@ -132,8 +132,8 @@ if [[ -z "$CC" ]]; then
     exit 1
 fi
 
-# Only lld version >= 9 can be considered stable
-if [[ "$compiler" == "clang" && "$majorVersion" -ge 9 ]]; then
+# Only lld version >= 9 can be considered stable. lld doesn't support s390x.
+if [[ "$compiler" == "clang" && "$majorVersion" -ge 9 && "$build_arch" != "s390x" ]]; then
     if "$CC" -fuse-ld=lld -Wl,--version >/dev/null 2>&1; then
         LDFLAGS="-fuse-ld=lld"
     fi

